src/mgmt/srv.c

```diff
@@ -300,6 +303,106 @@ process_join_request(Drpc__Call *drpc_req, Mgmt__JoinResp *resp)
 		resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
 }
 
+static void
+process_create_pool_request(Drpc__Call *drpc_req, Mgmt__CreatePoolResp *resp)
+{
+	Mgmt__CreatePoolReq	*pb_req = NULL;
+	d_rank_list_t		*targets = NULL;
+	d_rank_t		ranks[max_svc_nreplicas];
+	d_rank_list_t		svc = {};
+	d_rank_list_t		*svc_p = &svc;
+	uuid_t			pool_uuid;
+	int			buflen = 16;
+	int			len;
+	int			i;
+	int			rc = 0;
+	char			id[DAOS_UUID_STR_SIZE];
+	char			buf[buflen];
+	char			*svc_str = buf;
+	char			*extra = NULL;
+
+	memset(ranks, 0, sizeof(ranks));
+	svc_p->rl_ranks = ranks;
+
+	/* response status is populated with SUCCESS on init */
+	mgmt__create_pool_resp__init(resp);
+
+	/* Unpack the daos request from the drpc call body */
+	pb_req = mgmt__create_pool_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (pb_req == NULL) {
+		resp->status = MGMT__DAOS_REQUEST_STATUS__ERR_UNKNOWN;
+		D_ERROR("Failed to extract request\n");
+
+		return;
+	}
+
+	/* parse targets rank list */
+	if (strlen(pb_req->ranks) != 0) {
+		targets = daos_rank_list_parse(pb_req->ranks, ",");
+		if (targets == NULL) {
+			D_ERROR("failed to parse target ranks\n");
+			D_GOTO(out, rc = -DER_UNKNOWN);
+		}
+		D_DEBUG(DB_MGMT, "ranks in: %s\n", pb_req->ranks);
+	}
+
+	uuid_generate(pool_uuid);
+	D_DEBUG(DB_MGMT, DF_UUID": creating pool\n", DP_UUID(pool_uuid));
+
+	/* ranks to allocate targets (in) & svc for pool replicas (out) */
+	rc = ds_mgmt_create_pool(pool_uuid, pb_req->sys, "pmem",
+			targets, pb_req->scmbytes, pb_req->nvmebytes,
+			NULL /* props */, pb_req->numsvcreps, &svc_p);
+	if (targets != NULL)
+		d_rank_list_free(targets);
+	if (rc != 0) {
+		D_ERROR("failed to create pool: %d\n", rc);
+		goto out;
+	}
+
+	uuid_unparse_lower(pool_uuid, id);
+
+	if (svc_p->rl_nr == 0) {
+		D_ERROR("no pool svc replicas returned");
+		D_GOTO(out, rc = -DER_UNKNOWN);
+	}
+
+	/* Populate the pool service replica ranks string. */
+	snprintf(buf, buflen, "%u", svc_p->rl_ranks[0]);
+
+	for (i = 1; i < svc_p->rl_nr; i++) {
+		len = snprintf(svc_str, buflen, "%s,%u", svc_str,
+			svc_p->rl_ranks[i]);
+		if (len >= buflen) {
+			buflen *= 2;
+
+			extra = malloc(buflen * sizeof(char));
+			if (extra) {
+				snprintf(extra, buflen, "%s,%u", svc_str,
+						svc_p->rl_ranks[i]);
+				free(svc_str);
+				svc_str = extra;
+			} else {
+				D_ERROR("failed to allocate buffer");
+				D_GOTO(out, rc = -DER_NOMEM);
+			}
+		}
+	}
+
+	D_DEBUG(DB_MGMT, "%d service replicas: %s\n", svc_p->rl_nr, svc_str);
+	resp->uuid = id;
```

Comment:
id is in the stack; dRPC expects an allocated string, no?

Reply:
I think it is copied and allocated in mgmt__create_pool_resp__pack, with "body", is that correct? otherwise I imagine it wouldn't be working (which it is)